aten/src/ATen/core/dispatch/OperatorEntry.h

```diff
@@ -87,13 +94,30 @@ class OperatorEntry final {
   > kernels_;
   std::mutex kernelsMutex_; // protects kernels_
 
+  // unboxedAutogradKernels_ stores all autograd kernels registered for this op.
+  // An autograd kernel has the same signature as the main op kernel and
+  // internally re-dispatches to call the actual kernel.
+  // Autograd kernels are unboxed currently. We are planning to move this
+  // towards a system where ops register autograd wrappers (i.e. functions that
+  // do some wrapping code and get a pointer to the actual kernel) instead of
+  // autograd functions.
+  // This is a list because, similar to kernels_, multiple libraries could
+  // be loaded that register autograd kernels for the same op. The list is
+  // ordered by registration time descendingly, i.e. newer registrations are
+  // before older registrations and the list head is the autograd kernel
+  // which is currently used.
+  std::list<void*> unboxedAutogradKernels_;
+  std::mutex unboxedAutogradKernelsMutex_;
```

Comment:
again, I think it's fine to assume that library loads don't happen concurrently. Even if they did - having a top-level mutex is a much easier solution that also doesn't blow up OperatorEntry size.

btw, since you already have kernelsMutex_ above - why not just use that (even though we don't really need mutex)

Reply:
The mutex costs us even less, the complexity added by just the mutex it is negligible, but then we get parallel and race-free library loading for free, don't see a reason for not doing it.

The default pattern for mutexes is to bind them directly to the data they protect and trying to keep these data units small. OperatorEntry size doesn't matter, this is not an entry in the dispatch table.